runtime/sema/check_composite_declaration.go

```diff
@@ -1055,7 +1176,8 @@ func (checker *Checker) checkCompositeConformance(
 	// Ensure the composite kinds match, e.g. a structure shouldn't be able
 	// to conform to a resource interface
 
-	if interfaceType.CompositeKind != compositeType.Kind {
+	if interfaceType.CompositeKind != compositeType.Kind &&
+		interfaceType.CompositeKind != compositeType.getBaseCompositeKind() {
```

Comment:
Should this be a logical or?

Reply:
Thank you for the explanation, I see now the cases and what is meant to be checked.

So does this check assume that for an attachment, the first part will always fail?

Maybe it's clearer to differentiate more clearly between the "traditional" case (non-attachment), and the attachment case?

Reply:
Yes; an attachment's `kind` is always `CompositeKindAttachment`, while the interface's should either be `CompositeKindResource` or `CompositeKindStruct`